### PR TITLE
  Fix access so course staff can access inline analytics.

### PR DESCRIPTION
**Pull request by dcadams**
Currently course admins (instructors in the code) have the button but
    course staff (staff in the code) do not.

    This fix adds course staff to those who have access to the inline
    analytics buttton.

### CODE REVIEW

**Comment by dcadams** [comment]
@jbau or @stvstnfrd in your own time.

**Comment by jbau** [comment]
hmm. a bit surprised that the diff is so big, but that's just 1st impressions.

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 1]
```diff
@@ -1116,7 +1115,14 @@ def setUp(self):
         self.request = RequestFactory().get('/')
         self.request.user = self.user
         self.request.session = {}
-        self.course = CourseFactory.create()
+        self.course = CourseFactory.create(
+            org="A",
+            number="B",
+            run="C",
+        )
+        course_key = SlashSeparatedCourseKey.from_deprecated_string("A/B/C")
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_module_render.py]: Can't we grab the course_key directly from the `self.course` that we created above?

File: lms/djangoapps/courseware/tests/test_inline_analytics_integration.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ def setUp(self):
             run="C",
         )
         course_key = SlashSeparatedCourseKey.from_deprecated_string("A/B/C")
+        self.staff = StaffFactory(course_key=course_key)
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_inline_analytics_integration.py]: Can we remove the `course_key` variable and simply say:

```python
self.staff = StaffFactory(course_key=self.course.id)
```

**Comment by jbau** [comment]
:+1: from me, but see @stvstnfrd's suggestion

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 3]
```diff
@@ -1116,7 +1115,14 @@ def setUp(self):
...
         self.request.user = self.user
         self.request.session = {}
-        self.course = CourseFactory.create()
+        self.course = CourseFactory.create(
+            org="A",
+            number="B",
+            run="C",
+        )
+        course_key = self.course.id
+        self.staff = StaffFactory(course_key=course_key)
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_module_render.py]: Can we reference `course_key=self.course.id` like above?

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 4]
```diff
@@ -1138,7 +1144,18 @@ def setUp(self):
             self.user,
             self.descriptor
         )
+        self.field_data_cache_staff = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.staff,
+            self.descriptor
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_module_render.py]: Trailing comma ;)

File: lms/djangoapps/courseware/tests/test_module_render.py [hunk 5]
```diff
@@ -1138,7 +1144,18 @@ def setUp(self):
...
+        )
+        self.field_data_cache_instructor = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.instructor,
+            self.descriptor
```
Comment (stvstnfrd) [inline lms/djangoapps/courseware/tests/test_module_render.py]: Trailing comma ;)

**Comment by dcadams** [comment]
@stvstnfrd You're starting to go back over lines that no longer exist in my newer PR (#188). And at leas one comment is replicated.

Is it possible we can just merge this one and I'll fix up all the tiny stuff in the other one?

**Comment by stvstnfrd** [comment]
A few little things, otherwise +1.

**Comment by stvstnfrd** [comment]
Ah, my bad, must have had an unrefreshed page.

+1

**Comment by dcadams** [comment]
@jbau Tests passed this time. I believe the previous failure was due to my changes in the other in-line analytics PR (#188) causing /tmp/mako inconsistencies on Jenkins. Maybe you can purge /tmp after a run but no rush.

I'll merge this one now.

**Pull request merged**